tests/e2e/kepler_test.go

```diff
@@ -71,3 +73,122 @@ func TestBadKepler_Reconciliation(t *testing.T) {
 	ds := appsv1.DaemonSet{}
 	f.AssertNoResourceExists(exporter.DaemonSetName, components.Namespace, &ds)
 }
+
+func TestNodeSelector(t *testing.T) {
+	f := test.NewFramework(t)
+	// Ensure Kepler is not deployed (by any chance)
+	f.AssertNoResourceExists("kepler", "", &v1alpha1.Kepler{}, test.Timeout(10*time.Second))
+
+	nodes, err := f.GetResourceNames("node")
+	assert.NoError(t, err, "failed to get node names")
+	assert.NotZero(t, len(nodes), "got zero nodes")
+
+	node := nodes[0]
+	var labels k8s.StringMap = map[string]string{"e2e-test": "true"}
+	err = f.AddResourceLabels("node", node, labels)
+	assert.NoError(t, err, "could not label node")
+
+	f.CreateKepler("kepler", func(k *v1alpha1.Kepler) {
+		k.Spec.Exporter.Deployment.NodeSelector = labels.Merge(k8s.StringMap{
+			"kubernetes.io/os": "linux",
+		})
+	})
```

Comment:
We shouldn't need to add the `kubernetes.io/os: linux` label since the operator is expected to set it.

Reply:
changed